lib/galaxy/datatypes/binary.py

```diff
@@ -423,6 +432,8 @@ def set_meta( self, dataset, overwrite=True, **kwd ):
             dataset.metadata.read_groups = [ read_group['ID'] for read_group in dataset.metadata.bam_header.get( 'RG', [] ) if 'ID' in read_group ]
             dataset.metadata.sort_order = dataset.metadata.bam_header.get( 'HD', {} ).get( 'SO', None )
             dataset.metadata.bam_version = dataset.metadata.bam_header.get( 'HD', {} ).get( 'VN', None )
+            dataset.metadata.columns = 12
+            dataset.metadata.column_types = ['str', 'int', 'str', 'int', 'int', 'str', 'str', 'int', 'int', 'str', 'str', 'str']
```

Comment:
I think (I could be wrong) the main purpose is for accessing this sort of metadata in the tool wrappers without needing to call `str()` or `int()`. But since those are always the same I would prefer setting this directly as the default in the MetadataElement.